### PR TITLE
Configuration data and message logging enhancements.

### PR DESCRIPTION
**Pull request by toddsouthenbentley**
- Added `configData` member variable to ITMApplication that gets loaded during initialization.
- Added `ITMMessenger.isLoggingEnabled`.
- Added support for `ITMAPPLICATION_MESSAGE_LOGGING` and `ITMAPPLICATION_FULL_MESSAGE_LOGGING` config vars.

### CODE REVIEW

File: Sources/ITwinMobile/ITMApplication.swift [hunk 1]
```diff
@@ -99,6 +108,17 @@ open class ITMApplication: NSObject, WKUIDelegate, WKNavigationDelegate {
             }
             return Promise.value(())
         }
+        configData = loadITMAppConfig()
+        if let configData = configData {
+            extractConfigDataToEnv(configData: configData)
+            if configData.isYes("ITMAPPLICATION_MESSAGE_LOGGING") {
```
Comment (tcobbs-bentley) [inline Sources/ITwinMobile/ITMApplication.swift:114]: ITMAPPLICATION_MESSAGE_LOGGING and ITMAPPLICATION_FULL_MESSAGE_LOGGING should be documented in IOSSamples.xcconfig.

File: Sources/ITwinMobile/ITMApplication.swift [hunk 2]
```diff
@@ -270,9 +290,8 @@ open class ITMApplication: NSObject, WKUIDelegate, WKNavigationDelegate {
     }
 
     /// Loads the app config JSON from the main bundle.
-    /// Override this function in a subclass in order to add custom behavior.
     /// - Returns: The parsed contents of ITMAppConfig.json in the main bundle in the directory returned by `getWebAppDir`.
-    open func loadITMAppConfig() -> JSON? {
+    func loadITMAppConfig() -> JSON? {
```
Comment (tcobbs-bentley) [inline Sources/ITwinMobile/ITMApplication.swift:294]: I think we should still allow this to be overridden. I'm pretty sure SYNCHRO uses a different name for the json file. Don't forget to put the comment back also.

Comment (tcobbs-bentley) [reply Sources/ITwinMobile/ITMApplication.swift:294]: Or is that not allowed due to it being called from init?

Comment (tcobbs-bentley) [reply Sources/ITwinMobile/ITMApplication.swift:294]: If that's the case, then this probably needs to be an overridable class function.

File: Sources/ITwinMobile/ITMApplication.swift (hunk 1 above)
Comment (toddsouthenbentley) [reply Sources/ITwinMobile/ITMApplication.swift:114]: Yes, that file is in a different repo.

File: Sources/ITwinMobile/ITMApplication.swift (hunk 2 above)
Comment (toddsouthenbentley) [reply Sources/ITwinMobile/ITMApplication.swift]: Restored to previous definition and comment.

File: Sources/ITwinMobile/ITMApplication.swift (hunk 1 above)
Comment (tcobbs-bentley) [reply Sources/ITwinMobile/ITMApplication.swift:114]: Oops.

**Review by tcobbs-bentley — APPROVED** [review]

**Pull request merged**